rstar/src/algorithm/intersection_iterator.rs

```diff
@@ -50,13 +50,14 @@ where
             .iter()
             .filter(|c1| c1.envelope().intersects(&parent2.envelope()));
 
-        for child1 in children1 {
-            let children2 = parent2
-                .children()
-                .iter()
-                .filter(|c2| c2.envelope().intersects(&parent1.envelope()));
+        let children2: Vec<&RTreeNode<U>> = parent2
+            .children()
+            .iter()
+            .filter(|c2| c2.envelope().intersects(&parent1.envelope()))
+            .collect();
```

Comment:
I think it would be preferable to store the `Vec` inside the iterator itself and reuse the allocation instead of collecting a fresh one. (Depending on whether this is possible or not, another option might be to use a `SmallVec` to avoid heap allocations in some cases.)